src/webapp/forms.py

```diff
@@ -0,0 +1,74 @@
+import datetime
+from typing import List
+
+from flask_wtf import FlaskForm
+from wtforms import BooleanField, SelectField, SelectMultipleField, StringField, \
+    TimeField, HiddenField
+from wtforms.ext.dateutil.fields import DateField
+from wtforms.validators import InputRequired
+
+from webapp.common import gen_id
+
+
+class GenerateDowntimeForm(FlaskForm):
+    scheduled = BooleanField("Scheduled", None)  # default="checked" does not do anything
+    severity = SelectField("Severity", [InputRequired()], choices=[
+        ("Outage", "Outage"),
+        ("Severe", "Severe"),
+        ("Intermittent Outage", "Intermittent Outage"),
+        ("No Significant Outage Expected", "No Significant Outage Expected")
+    ])
+    description = StringField("Description", [InputRequired()])
+    start_date = DateField("Start Date", [InputRequired()])
+    start_time = TimeField("Start Time", [InputRequired()])
+    end_date = DateField("End Date", [InputRequired()])
+    end_time = TimeField("End Time", [InputRequired()])
+    services = SelectMultipleField("Services (select one or more)", [InputRequired()], choices=[])
+    resource = HiddenField("Resource")
+
+    class Meta:
+        csrf = False  # CSRF not needed because no data gets modified
+
+    # https://stackoverflow.com/a/21815180
+    def validate(self):
+        if not super().validate():
+            return False
+        start_datetime = datetime.datetime.combine(self.start_date.data, self.start_time.data)
+        end_datetime = datetime.datetime.combine(self.end_date.data, self.end_time.data)
+        if start_datetime >= end_datetime:
+            self.end_time.errors.append("End date/time must be after start date/time")
```

Comment:
Can we put any input error messages on the right, in the same area that the yaml would get generated? Also, making them an angry red would make them more readily visible

Reply:
That would be pretty hard to do -- the error messages are formatted by the framework. I can do the angry red though.